### PR TITLE
Secures password using JBCrypt and fixes broken tests

### PR DESCRIPTION
**Pull request by joycezhao2**
Adds JBCrypt dependency in pom.xml,  generates password hash in RegisterServlet, checks hashed password in LoginServlet, generates hashed passwords in DefaultDataStore. Fixes broken tests in RegisterServletTest and LoginServletTest

### CODE REVIEW

File: src/main/java/codeu/controller/LoginServlet.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.mindrot.jbcrypt.BCrypt;
```
Comment (swarelle) [inline src/main/java/codeu/controller/LoginServlet.java]: I think it's convention to leave an extra line between imports from different packages?

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 2]
```diff
@@ -8,9 +8,9 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
 import codeu.model.data.User;
 import codeu.model.store.basic.UserStore;
+import org.mindrot.jbcrypt.BCrypt;
```
Comment (swarelle) [inline src/main/java/codeu/controller/RegisterServlet.java]: Same here for the newline? And also maybe organize the imports alphabetically!

File: src/main/java/codeu/model/store/basic/DefaultDataStore.java [hunk 3]
```diff
@@ -23,6 +23,7 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.UUID;
+import org.mindrot.jbcrypt.BCrypt;
```
Comment (swarelle) [inline src/main/java/codeu/model/store/basic/DefaultDataStore.java]: Newline?

File: src/test/java/codeu/controller/LoginServletTest.java [hunk 4]
```diff
@@ -30,6 +30,7 @@
 import org.junit.Test;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
+import org.mindrot.jbcrypt.BCrypt;
```
Comment (swarelle) [inline src/test/java/codeu/controller/LoginServletTest.java]: Newline?

File: src/test/java/codeu/controller/RegisterServletTest.java [hunk 5]
```diff
@@ -15,6 +15,7 @@
 
 import codeu.model.data.User;
 import codeu.model.store.basic.UserStore;
+import org.mindrot.jbcrypt.BCrypt;
```
Comment (swarelle) [inline src/test/java/codeu/controller/RegisterServletTest.java]: Newline?

File: src/main/java/codeu/controller/LoginServlet.java (hunk 1 above)
Comment (joycezhao2) [reply src/main/java/codeu/controller/LoginServlet.java]: got it... thanks!

**Pull request merged**